PL2/PL2-core/src/main/java/nl/tudelft/pl2016gr2/core/algorithms/SplitGraphs.java

```diff
@@ -0,0 +1,151 @@
+package nl.tudelft.pl2016gr2.core.algorithms;
+
+import nl.tudelft.pl2016gr2.model.Node;
+import nl.tudelft.pl2016gr2.model.OriginalGraph;
+import nl.tudelft.pl2016gr2.test.utility.TestId;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Provides functionality for extracting a subgraph of a sequence graph based on a subset of
+ * genomes.
+ * <p>
+ * The resulting subgraph will provide the following characteristics.
+ * <ul>
+ * <li>The nodes in the subgraph are a subset of the nodes in the original graph.</li>
+ * <li>The nodes in the subgraph will have an identical ID as their counterparts in the original
+ * graph.</li>
+ * <li>The nodes will be newly instantiated, meaning that the subgraph datastructure is not backed
+ * by the original graph.</li>
+ * </ul>
+ * </p>
+ *
+ * @author Wouter Smit
+ */
+public class SplitGraphs {
+
+  @TestId(id = "graph_field")
+  private final OriginalGraph mainGraph;
+
+  /**
+   * Instantiates an algorithmic class on a <code>OriginalGraph</code>.
+   * <p>
+   * When {@link #getSubgraph(Collection)} is called, the class will construct a new instance of
+   * the
+   * graph containing only the specified genomes.
+   * </p>
+   *
+   * @param mainGraph The graph to split
+   */
+  public SplitGraphs(OriginalGraph mainGraph) {
```

Comment:
I did not do so because the `GraphInterface` does not specify genome information (`AbstractNode`s do not contain genome information). OriginalGraph is the first object in the graph inheritance that provides this functionality.